lib/daru/io/importers/excelx.rb

```diff
@@ -0,0 +1,132 @@
+        # @param skiprows [Integer] Skips the first +:skiprows+ number of rows from the
+        #   sheet being parsed.
+        # @param skipcols [Integer] Skips the first +:skipcols+ number of columns from the
+        #   sheet being parsed.
+        # @param order [Boolean] Defaults to true. When set to true, first row of the
+        #   given sheet is used as the order of the Daru::DataFrame and data of
+        #   the Dataframe consists of the remaining rows.
+        # @param index [Boolean] Defaults to false. When set to true, first column of the
+        #   given sheet is used as the index of the Daru::DataFrame and data of
+        #   the Dataframe consists of the remaining columns.
+        #
+        #   When set to false, a default order (0 to n-1) is chosen for the DataFrame,
+        #   and the data of the DataFrame consists of all rows in the sheet.
+        #
+        # @return A +Daru::DataFrame+ imported from the given XLSX file and sheet
+        #
+        # @example Importing from a local file
+        #   path  = 'spec/fixtures/excelx/Stock-counts-sheet.xlsx'
+        #   sheet = 'Example Stock Counts'
+        #   df    = Daru::IO::Importers::XLSX.new(path, sheet: sheet).call
+        #   df
+        #
+        #   #=> <Daru::DataFrame(15x7)>
+        #              Status Stock coun  Item code        New Descriptio Stock coun Offset G/L
+        #        0          H          1        nil        nil New stock  2014-08-01        nil
+        #        1        nil          1  IND300654          2 New stock  2014-08-01      51035
+        #        2        nil          1   IND43201          5 New stock  2014-08-01      51035
+        #        3        nil          1   OUT30045          3 New stock  2014-08-01      51035
+        #       ...       ...        ...     ...           ...     ...       ...           ...
+        #
+        # @example Importing from a local file without headers
+        #   path  = 'spec/fixtures/excelx/Stock-counts-sheet.xlsx'
+        #   sheet = 'Example Stock Counts'
+        #   df    = Daru::IO::Importers::XLSX.new(path, sheet: sheet, headers: false).call
+        #   df
+        #
+        #   #=> <Daru::DataFrame(16x7)>
+        #                   0           1          2          3          4          5        6
+        #        0      Status Stock coun  Item code        New Descriptio Stock coun Offset G/L
+        #        1          H          1        nil        nil New stock  2014-08-01        nil
+        #        2        nil          1  IND300654          2 New stock  2014-08-01      51035
+        #        3        nil          1   IND43201          5 New stock  2014-08-01      51035
+        #        4        nil          1   OUT30045          3 New stock  2014-08-01      51035
+        #       ...       ...        ...     ...           ...     ...       ...           ...
+        #
+        # @example Importing from a remote URL
```

Comment:
How will this documentation be easily accessible to someone who is using the `from_excel` method? This implementation of using importer and exporter classes is opaque to the user, right?